samples/matter/common/src/binding_handler.cpp

```diff
@@ -0,0 +1,148 @@
+/*
+ * Copyright (c) 2023 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include "binding_handler.h"
+
+#include <zephyr/logging/log.h>
+
+LOG_MODULE_DECLARE(app, CONFIG_CHIP_APP_LOG_LEVEL);
+
+using namespace chip;
+using namespace chip::app;
+
+CHIP_ERROR BindingHandler::Init(DeviceChangedHandler handler)
+{
+	VerifyOrReturnError(mDeviceChangedHandler != nullptr, CHIP_ERROR_INVALID_ARGUMENT);
+	InitInternal();
+	
+	return CHIP_NO_ERROR;
+}
+
+void BindingHandler::OnInvokeCommandSucces()
+{
+	LOG_DBG("Binding command applied successfully!");
+
+	/* If session was recovered and communication works, reset flag to the initial state. */
+	if (BindingHandler::Instance().mCaseSessionRecovered)
+		BindingHandler::Instance().mCaseSessionRecovered = false;
+}
+void BindingHandler::OnInvokeCommandFailure(BindingData &bindingData, CHIP_ERROR aError)
+{
+	CHIP_ERROR error;
+
+	if (aError == CHIP_ERROR_TIMEOUT && !BindingHandler::Instance().mCaseSessionRecovered) {
+		LOG_INF("Response timeout for invoked command, trying to recover CASE session.");
+
+		/* Set flag to not try recover session multiple times. */
+		BindingHandler::Instance().mCaseSessionRecovered = true;
+
+		/* Allocate new object to make sure its life time will be appropriate. */
+		BindingHandler::BindingData *data = Platform::New<BindingHandler::BindingData>();
+		VerifyOrReturn(data != nullptr, LOG_ERR("New object allocation failed"););
```

Comment:
I'm not 100% sure, but I think `;` is not needed.
```suggestion
		VerifyOrReturn(data != nullptr, LOG_ERR("New object allocation failed"));
```